source/compiler-core/slang-dxc-compiler.cpp

```diff
@@ -567,7 +567,7 @@ SlangResult DXCDownstreamCompiler::compile(const CompileOptions& inOptions, IArt
     DxcIncludeHandler includeHandler(&searchDirectories, options.fileSystemExt, options.sourceManager);
 
     ComPtr<IDxcOperationResult> dxcOperationResult;
-    SLANG_RETURN_ON_FAIL(dxcCompiler->Compile(dxcSourceBlob,
+    SLANG_RETURN_ON_FAIL(SlangResult(dxcCompiler->Compile(dxcSourceBlob,
```

Comment:
Instead of changing every use to cast to SlangResult, can we change SLANG_RETURN_ON_FAIL so it handles both SlangResult and other integer return code types?

Reply:
I've added SLANG_RETURN_ON_FAIL_HRESULT, to retain some type safety.

(This is actually obviated by the other change making SlangResult not an enum class)